# HG changeset patch
# User Thorsten Meyer <thorsten.meyier@gmx.de>
# Date 1298125961 -3600
#      Sat Feb 19 15:32:41 2011 +0100
# Node ID b3f9f98e7640a8ed2a061dd39e35bbcd1baf8e66
# Parent  68daf2a0b56ded488e8f183d415d0c6026166b17
Fix interp1 tests according to spline update of 07e102029d2a

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2011-02-19  Thorsten Meyer  <thorsten.meyier@gmx.de>
+
+	* general/interp1.m: Fix tests according to spline update of
+	changeset 07e102029d2a (see bug 31780)
+
 2011-02-17  Kai Habel  <kai.habel@gmx.de>
 
 	* plot/private/__add_default_menu__.m: Rename fltk_gui_mode here
 	as well.
 
 2011-02-16  Ben Abbott  <bpabbott@mac.com>
 
 	* plot/subplot.m: Decrease spacing between subplots rows / columns.
diff --git a/scripts/general/interp1.m b/scripts/general/interp1.m
--- a/scripts/general/interp1.m
+++ b/scripts/general/interp1.m
@@ -592,25 +592,25 @@ endfunction
 %!error interp1
 %!error interp1(1:2,1:2,1,"bogus")
 
 %!assert (interp1(1:2,1:2,1.4,"nearest"),1);
 %!error interp1(1,1,1, "linear");
 %!assert (interp1(1:2,1:2,1.4,"linear"),1.4);
 %!error interp1(1:3,1:3,1, "cubic");
 %!assert (interp1(1:4,1:4,1.4,"cubic"),1.4);
-%!error interp1(1:2,1:2,1, "spline");
+%!assert (interp1(1:2,1:2,1.1, "spline"), 1.1);
 %!assert (interp1(1:3,1:3,1.4,"spline"),1.4);
 
 %!error interp1(1,1,1, "*nearest");
 %!assert (interp1(1:2:4,1:2:4,1.4,"*nearest"),1);
 %!error interp1(1,1,1, "*linear");
 %!assert (interp1(1:2:4,1:2:4,[0,1,1.4,3,4],"*linear"),[NA,1,1.4,3,NA]);
 %!error interp1(1:3,1:3,1, "*cubic");
 %!assert (interp1(1:2:8,1:2:8,1.4,"*cubic"),1.4);
-%!error interp1(1:2,1:2,1, "*spline");
+%!assert (interp1(1:2,1:2,1.3, "*spline"), 1.3);
 %!assert (interp1(1:2:6,1:2:6,1.4,"*spline"),1.4);
 
 %!assert (interp1([3,2,1],[3,2,2],2.5),2.5)
 
 %!assert (interp1 ([1,2,2,3,4],[0,1,4,2,1],[-1,1.5,2,2.5,3.5], "linear", "extrap"), [-2,0.5,4,3,1.5])
 %!assert (interp1 ([4,4,3,2,0],[0,1,4,2,1],[1.5,4,4.5], "linear"), [0,1,NA])
 %!assert (interp1 (0:4, 2.5), 1.5)
